KalturaMediaServer/src/com/kaltura/media/server/api/KalturaWebServicesServer.java

```diff
@@ -42,11 +44,12 @@ public Endpoint addService(IWebService service)
 	}
 
 	public void shutdown() {
-		Set<String> keys = services.keySet();
-		for(String key : keys)
+		Iterator<Map.Entry<String, Endpoint>> serviceIter = services.entrySet().iterator();
```

Comment:
In the next section we can see the services.remove(key); which removes the entry , did not get what you ment